src/libs/sfdk/dockervirtualmachine.cpp

```diff
@@ -206,77 +199,52 @@ void DockerVirtualMachinePrivate::start(const QObject *context, const Functor<bo
     Q_ASSERT(context);
     Q_ASSERT(functor);
 
-    const QPointer<const QObject> context_{context};
+    BatchComposer composer = BatchComposer::createBatch("DockerVirtualMachinePrivate::start");
+    composer.batch()->setPropagateFailure(true);
```

Comment:
It seems to me that createBatch() is really often followed by composer.batch()->setPropagateFailure(true). Perhaps you could consider having an alternative version of createBatch, which takes propagateFailure as parameter?